File: Binary Trees/Traversals/Inorder Traversal/iterative.cpp
class Solution {
    public:
        vector<int> inorderTraversal(TreeNode* root) {
            if(root == nullptr) return {};
            vector<int> ans;
            stack<TreeNode*> st;
            TreeNode* currentNode = root;
            while(true) {
                if(currentNode != nullptr){
                    st.push(currentNode);
                    currentNode = currentNode->left;
                }
                else{
                    if(st.empty()) break;
                    currentNode = st.top();
                    st.pop();
                    ans.push_back(currentNode->val);
                    currentNode = currentNode->right;
                }
            }
            return ans;
        }
    };

File: Binary Trees/Height of Binary Tree/iterative.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int maxDepth(TreeNode root) {
        if (root == null) return 0;
        Stack<Pair<TreeNode, Integer>> stack = new Stack<>();
        stack.push(new Pair<>(root, 1));
        int maxDepth = 0;
        while (!stack.isEmpty()) {
            Pair<TreeNode, Integer> current = stack.pop();
            TreeNode node = current.getKey();
            int depth = current.getValue();
            if (node != null) {
                maxDepth = Math.max(maxDepth, depth);
                stack.push(new Pair<>(node.left, depth + 1));
                stack.push(new Pair<>(node.right, depth + 1));
            }
        }
        return maxDepth;
    }
}